GitUI/CommandsDialogs/BrowseDialog/FormCommitCount.cs

```diff
@@ -28,69 +35,113 @@ private void cbIncludeSubmodules_CheckedChanged(object sender, EventArgs e)
 
         private void FetchData()
```

Comment:
:memo: In general, before merge we'd want to propagate this as an asynchronous method rather than leave it as fire-and-forget. I'm not sure how difficult that is for this method.

Reply:
It's called from two different event handlers. What's best practice in such cases?

Eg: https://github.com/drewnoakes/gitextensions/blob/c3750c9b492ebf56a626935d36301cd74b125baa/GitUI/CommandsDialogs/BrowseDialog/FormCommitCount.cs#L28

Reply:
> What's best practice in such cases?

You want to move the async entry point as close to the point of API limitations as possible. In this case, the event handlers would be synchronous but `FetchDataAsync` would be asynchronous.

Reply:
Makes sense in general. In this case I think it'd create more duplicated code. Take a the most recent commit and see if you'd still do it differently.

Reply:
> Take a the most recent commit and see if you'd still do it differently.

Yes, I would still push the change all the way to the true boundary.

Reply:
Done.